pallets/author-noting/src/test.rs

```diff
@@ -445,3 +444,121 @@ fn test_set_author() {
             );
         });
 }
+
+#[test]
+#[should_panic(expected = "DidSetContainerAuthorData must be updated only once in a block")]
+fn test_on_initalize_does_not_kill_and_panics() {
+    BlockTests::new()
+        .skip_author_noting_on_initialize()
+        .with_relay_sproof_builder(|_, relay_block_num, sproof| match relay_block_num {
+            1 => {
+                crate::DidSetContainerAuthorData::<Test>::put(true);
+                let slot: InherentType = 13u64.into();
+                let mut s = ParaHeaderSproofBuilderItem::default();
+                s.para_id = 1001.into();
+                s.author_id =
+                    HeaderAs::NonEncoded(sp_runtime::generic::Header::<u32, BlakeTwo256> {
+                        parent_hash: Default::default(),
+                        number: Default::default(),
+                        state_root: Default::default(),
+                        extrinsics_root: Default::default(),
+                        digest: sp_runtime::generic::Digest {
+                            logs: vec![DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode())],
+                        },
+                    });
+                sproof.items.push(s);
+            }
+            _ => unreachable!(),
+        })
+        .add(1, || {
+            assert_eq!(AuthorNoting::latest_author(ParaId::from(1001)), Some(13u64));
```

Comment:
This `add(` is not needed, right?

Reply:
actually the add is needed but can be empty